server/questions.js

```diff
@@ -52,12 +52,17 @@ exports.addQuestionByType = function(qType, question, callback) {
 
 		case common.questionTypes.MULTIPLECHOICE.value:
 			questionToAdd.type = common.questionTypes.MULTIPLECHOICE.value;
-			questionToAdd.choices = question.choices;
+			questionToAdd.choices = question.choices ? question.choices : [];
```

Comment:
validate at the beginning of the function. No need for this change

Reply:
I have to check for it in the question type it self, thats why I have those switch statements

Reply:
But the validation will take care of checking for you

Reply:
right... we can have all the validation done in that one function